Sources/_NIOConcurrency/AsyncAwaitSupport.swift

```diff
@@ -0,0 +1,143 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+extension NIO.System {
+    public static var hasAsyncAwaitSupport: Bool {
+        #if compiler(>=5.4) && $AsyncAwait
+        return true
+        #else
+        return false
+        #endif
+    }
+}
+
+#if compiler(>=5.4) && $AsyncAwait
+import NIO
+
+extension EventLoopFuture {
+    /// Get the value/error from an `EventLoopFuture` in an `async` context.
+    ///
+    /// This function can be used to bridge an `EventLoopFuture` into the `async` world. Ie. if you're in an `async`
+    /// function and want to get the result of this future.
+    public func get() async throws -> Value {
+        return try await withUnsafeThrowingContinuation { cont in
+            self.whenComplete { result in
+                switch result {
+                case .success(let value):
+                    cont.resume(returning: value)
+                case .failure(let error):
+                    cont.resume(throwing: error)
+                }
+            }
+        }
+    }
+}
+
+extension EventLoopPromise {
+    /// Complete a future with the result (or error) of the `async` function `body`.
+    ///
+    /// This function can be used to bridge the `async` world into an `EventLoopPromise`.
+    ///
+    /// - parameters:
+    ///   - body: The `async` function to run.
+    public func completeWithAsync(_ body: @escaping () async throws -> Value) {
+        Task.runDetached {
+            do {
+                let value = try await body()
+                self.succeed(value)
+            } catch {
+                self.fail(error)
+            }
+        }
+    }
+}
+
+extension Channel {
+    public func writeAndFlush<T>(_ any: T) async throws {
```

Comment:
Is there a reason this is defined on `Channel` instead of `ChannelOutboundInvoker`?

Reply:
good catch, don't think so :)

Reply:
turns out yes: For some reason, in NIO `Channel` has the `<T>wAF(T)` and `ChannelOutboundInvoker` only has the `NIOAny` version... Mirrored that here.